filament/src/FrameInfo.h

```diff
@@ -130,6 +131,50 @@ class CircularQueue {
     using reference = value_type&;
     using const_reference = value_type const&;
 
+    class iterator {
```

Comment:
Could we make the iterator a templated class based on the pointer type (either `T*` or `const T*`)? This way we don't have duplication between `iterator` and `const_iterator`:

```
template<typename Ptr>
class iterator {
    ...
    using pointer = Ptr;
    ...
};
```